### PR TITLE
Add config class for SplittingTransformer

### PR DESCRIPTION
**Pull request by davidh44**
## Modifications

Add `SplitTransformerConfiguration` for when calling `AsyncResponseTransformer.split()`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    SplitTransformerConfiguration> {
+
+    private final Long bufferSize;
+
+    private SplitTransformerConfiguration(DefaultBuilder builder) {
+        this.bufferSize = Validate.paramNotNull(builder.bufferSize, "bufferSize");
+    }
+
+    /**
+     * Create a {@link FileRequestBodyConfiguration.Builder}, used to create a {@link SplitTransformerConfiguration}.
```
Comment (dagnir) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java:43]: Why does this mention `FileRequestBodyConfiguration`?

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.internal.async.SplittingTransformer;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configuration options for {@link AsyncResponseTransformer#split(SplitTransformerConfiguration)} to configure how the SDK
+ * should split the {@link AsyncResponseTransformer}.
+ *
```
Comment (dagnir) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java:29]: This description makes me thing the name is not quite right. "to configure __how__ the SDK should split..." sounds like it's configuration for the splitting transformer (the one we're calling `split` on), but the name makes it sound like it's for the `SplitAsyncResponseTransformer`, which is the result of the `split`.

@zoewangg WDYT?

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java (hunk 1 above)
Comment (davidh44) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java]: oops missed deleting that

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java (hunk 2 above)
Comment (davidh44) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java:29]: right, its used to instantiate a `SplittingTransformer` which is then passed to the `SplitAsyncResponseTransformer`

`SplittingTransformerConfiguration`?

Comment (dagnir) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java:29]: Okay thanks for clarifying. In that case then yeah I vote for `SplittingTransformerConfiguration`

**Review by dagnir — APPROVED** [review]

**Pull request merged**